src/pybind/mgr/rbd_support/mirror_snapshot_schedule.py

```diff
@@ -369,6 +369,9 @@ def init_schedule_queue(self) -> None:
         self.queue: Dict[str, List[ImageSpec]] = {}
         # pool_id => {namespace => image_id}
         self.images: Dict[str, Dict[str, Dict[str, str]]] = {}
+        # Previous versions incorrectly stored the global config in
+        # the localized module option. Check the config is here and fix it.
+        Schedules.replace_local_with_global_mod_option(self)
```

Comment:
It does not look very nice to me. I would prefer if we had the schedules initialization here (`self.schedules =  Schedules(self)`), where this would be called. And then in `load_schedules` just `self.schedules.load(...)`. And the schedules' `load` function might need updating to reset the schedules before loading the new ones.

Reply:
sounds good. I'll try this out.

Reply:
Done. Let me know if it looks okay.